src/routes/sessions/[id]/+page.svelte

```diff
@@ -410,6 +391,33 @@
 	}
 
 	.prompt-editor__toolbar {
-		@apply flex items-center gap-x-2;
+		@apply flex items-stretch gap-x-2;
+	}
+
+	.prompt-editor__stop {
+		@apply relative -mx-3 -my-2 h-9 w-9;
+	}
+
+	.prompt-editor__stop:hover {
+		.prompt-editor__stop-icon {
+			@apply opacity-100;
+		}
+
+		.prompt-editor__loading-icon {
+			@apply opacity-0;
+		}
+	}
+
+	.prompt-editor__stop-icon,
+	.prompt-editor__loading-icon {
+		@apply absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2;
+	}
+
+	.prompt-editor__stop-icon {
+		@apply opacity-0;
+	}
+
+	.prompt-editor__loading-icon {
+		@apply opacity-100;
 	}
 </style>
```

Comment:
Adds a loading spinner while completion is in progress:


https://github.com/user-attachments/assets/8658ea7b-cacf-4570-a0b7-c80b05e5d8d6